src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -12,10 +12,36 @@ public ChessBoard() {
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+        int numberOfWhitePawns = 0;
+        int numberOfBlackPawns = 0;
+
+        for (Pawn[] col : pieces) {
+            for (Pawn row : col) {
+                if (row != null) {
+                    if (pieceColor.equals(PieceColor.WHITE)) numberOfWhitePawns += 1;
+                    else numberOfBlackPawns += 1;
+                }
+            }
+        }
+
+        if (!validNrOfPawns(pieceColor, numberOfWhitePawns, numberOfBlackPawns)
+        || !isLegalBoardPosition(pawn.getXCoordinate(), pawn.getYCoordinate())
+        || pieces[xCoordinate][yCoordinate] != null) {
+            pawn.setXCoordinate(-1);
+            pawn.setYCoordinate(-1);
+
```

Comment:
return here, so you can get rid of the else, it's less complicated to follow the logic this way